### DISCUSSION TITLE
minting-dapp whitelist

### DISCUSSION THREAD

**Discussion by zueyong123** [hashlips-lab/nft-erc721-collection#227] [Q&A]
Hi everyone, I would really appreciate it if you could help me with these few questions I have 
1) Will I be able to update the whitelist wallets after I've deployed the contract on the mainnet if so do I just insert them into the whietlist.json and save it
2) How would I go about making the preview image fill out the whole box I also used a gif instead of a png I change the gif to png and it still worked, however, I'm not sure if this may cause issues in the future.
![image](https://user-images.githubusercontent.com/97086494/162778862-789c6e18-87ce-4072-9356-8596692bec30.png)

Thank you for your help !!!!

**Comment by m360ai** [comment]
As far as I know, 
1. Yes you can update the JSON and re-create the whitelist after using yarn start/build. You should test that feature on Rinkeby first! Test everything first on rinkeby. 
2. You need to change the CSS to get the div or element to resize for the image to resize. Try to use the same format for the image (I.E. PNG) as in the original code if you're unsure.

**Reply by liarco** [reply]
@m360ai promoting private support via e-mail is considered spam. Unfortunately bad users are using this method to scam people. Not saying you are one of them, but I must block any attempt for safety.

If you are willing to support, please feel free to use this public place like we all do.

Thank you for your understanding.

Also, unfortunately the first answer is wrong: the whitelist can be updated as many times as you want. Please double-check before telling a user to redeploy his contract, it can be a pretty expensive mistake.

**Comment by liarco** [comment]
@zueyong123 here are the required steps to update the whitelist:
1. as long as you have a deployed contract, it doesn't matter if the whitelist sale is already open or not, the following steps always work
2. update your `whitelist.json` file
3. rebuild the dapp with `yarn build` inside the `minting-dapp` folder and redeploy it to you hosting platform
4. run the `yarn whitelist-open --network YOUR_NETWORK` inside the `smart-contract` folder in order to update the Merkle Tree root hash (you can wait until you really need to open the sale)

**Reply by FrontlineHeroesToken** [reply]
i have done these steps, but it doesnt update whitelist wallet address. should it produce a new root hash when these steps are followed?

**Reply by FrontlineHeroesToken** [reply]
disregard. it is working. thanks for this. makes adding tiers easier..

**Reply by omniNerd** [reply]
> @zueyong123 here are the required steps to update the whitelist:
> 
> 1. as long as you have a deployed contract, it doesn't matter if the whitelist sale is already open or not, the following steps always work
> 2. update your `whitelist.json` file
> 3. rebuild the dapp with `yarn build` inside the `minting-dapp` folder and redeploy it to you hosting platform
> 4. run the `yarn whitelist-open --network YOUR_NETWORK` inside the `smart-contract` folder in order to update the Merkle Tree root hash (you can wait until you really need to open the sale)

Hi Liarco, 

Thanks so much for this repository. I'm very thankful for your hard work. I am having an issue updating the whitelist. I followed all the directions above and the new addresses are still not able to mint during the whitelist sale. The contract has already been deployed to the mainnet, I'm using version 2.3.0, and I have rebuilt the dApp several times after updating the whitelist json file. For some reason the new addresses aren't able to mint during the whitelist sale, but the whitelisted addresses that were deployed with the contract work fine. If there is any advice you can give me I would greatly appreciate considering our launch is only a few days away. Here is a link to the contract on Etherscan:

https://etherscan.io/address/0xd3ea7249fd1e00fffd8619aeb50c7f9090bf0dc1#code

Many Thanks,
Peter

**Comment by m360ai** [comment]
Hi Marco, this is great! I'm very happy to hear this because I actually
didn't know it was possible while keeping the same contract. I originally
thought it needed to be a function which would need to be interacted with
on the ERC721A. I know it would be expensive so that's why is said to test
it o. Rinkeby first but anyway thank you for the clarification. Much
appreciated.

I have a question about the public vs. Pre-sale. I've interacted with the
contract directly and been able to pause, unpause, update price etc but can
you differentiate between public sale or presale when doing this?

Thank you.

On Mon, Apr 11, 2022 at 1:43 PM Marco Lipparini ***@***.***>
wrote:

> @zueyong123 <https://github.com/zueyong123> here are the required steps
> to update the whitelist:
>
>    1. as long as you have a deployed contract, it doesn't matter if the
>    whitelist sale is already open or not, the following steps always work
>    2. update your whitelist.json file
>    3. rebuild the dapp with yarn build inside the minting-dapp folder and
>    redeploy it to you hosting platform
>    4. run the yarn whitelist-open --network YOUR_NETWORK inside the
>    smart-contract folder in order to update the Merkle Tree root hash
>    (you can wait until you really need to open the sale)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hashlips-lab/nft-erc721-collection/discussions/227#discussioncomment-2547576>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5UXMXWNVBBW74VFGRLQ7DVESFFNANCNFSM5TDSVFCQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <hashlips-lab/nft-erc721-collection/repo-discussions/227/comments/2547576@
> github.com>
>
-- 
Respectfully,
Michael Croft M.A.
213.944.2742

**Reply by liarco** [reply]
Hi @m360ai,
to be honest I think I misunderstood the original question because I thought the contract was already deployed to mainnet, that's why I was so worried about suggesting to redeploy. But now I see it was a question for a future deployment so it's my fault here!

>  have a question about the public vs. Pre-sale. I've interacted with the
contract directly and been able to pause, unpause, update price etc but can
you differentiate between public sale or presale when doing this?

There is no real difference between `pre-sale` and `public sale` (from a technical point of view). I decided to create dedicated configuration for those two sale steps because they are really common in NFT projects, so setting those values from the beginning and having dedicated commands to set everything up on the contract makes it easier for unexperienced users to run everything without making mistakes.
If you take a look at the scripts for opening and closing those sales (e.g. [4_presale_open.ts](https://github.com/hashlips-lab/nft-erc721-collection/blob/main/smart-contract/scripts/4_presale_open.ts) or [6_public_sale_open.ts](https://github.com/hashlips-lab/nft-erc721-collection/blob/main/smart-contract/scripts/6_public_sale_open.ts)) you will see that they are exactly the same except for the configuration values they are using.

I hope this answers the question. 😉

**Comment by m360ai** [comment]
Hi, that's great news! Just wanted to let you know that I recently deployed
Https://www.thenftsforukraine.com
with the Hashlips DAPP and ERC721A. It is a partial benefit collection. I
launched in three stages, whitelist, pre-sale, public with a hidden
Metadata image pre-reveal, and all has worked perfectly.
Check it out and let me know what you think.


On Thu, Apr 14, 2022 at 9:06 AM FrontlineHeroesToken <
***@***.***> wrote:

> disregard. it is working. thanks for this. makes adding tiers easier..
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hashlips-lab/nft-erc721-collection/discussions/227#discussioncomment-2567963>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM5UXMUZTLVMDSALMVVT7JLVFA66TANCNFSM5TDSVFCQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <hashlips-lab/nft-erc721-collection/repo-discussions/227/comments/2567963@
> github.com>
>
-- 
Respectfully,
Michael Croft M.A.
213.944.2742
